packages/xstate-store/src/react.ts

```diff
@@ -94,3 +100,48 @@ export const useStore: {
 
   return storeRef.current;
 };
+
+/**
+ * A React hook that subscribes to the `atom` and returns the current value of
+ * the atom.
+ *
+ * @example
+ *
+ * ```ts
+ * const atom = createAtom(0);
+ *
+ * const Component = () => {
+ *   const count = useAtom(atom);
+ *
+ *   return (
+ *     <div>
+ *       <div>{count}</div>
+ *       <button onClick={() => atom.set((c) => c + 1)}>Increment</button>
+ *     </div>
+ *   );
+ * };
+ * ```
+ *
+ * @param atom The atom, created from `createAtom(…)`
+ * @param selector An optional function which takes in the `snapshot` and
+ *   returns a selected value
+ * @param compare An optional function which compares the selected value to the
+ *   previous value
+ */
+export function useAtom<TAtom extends AnyAtom>(
+  atom: TAtom
+): SnapshotFromAtom<TAtom>;
```

Comment:
for such easy utils, I feel this is just an anti-pattern - we are using it in other places because the alternative is just annoying but when u can avoid surfacing conditional types in the public signature, you should avoid it. Conditional types stay deferred at the generic level and makes wrapping functions using them much harder.

In here, we should strive for a simple:
```ts
export function useAtom<T>(
  atom: Atom<T>
): T
export function useAtom<T, S>(
  atom: Atom<T>,
  selector: (snapshot: T) => S,
  compare?: (a: S, b: S) => boolean
): S
```

But I guess we are using the proposed pattern already in some places so it might be more consistent to just keep it 🤷